iib/web/models.py

```diff
@@ -190,6 +209,101 @@ def add_state(self, state, state_reason):
         db.session.flush()
         self.request_state_id = request_state.id
 
+    def to_json(self, verbose=True):
+        def _state_to_json(state):
+            return {
+                'state': RequestStateMapping(state.state).name,
+                'state_reason': state.state_reason,
+                'updated': state.updated.isoformat() + 'Z',
+            }
+
+        rv = {
+            'id': self.id,
+            'arches': [arch.name for arch in getattr(self.index_image, 'architectures', [])],
+            'binary_image': self.binary_image.pull_specification,
+            'binary_image_resolved': getattr(
+                self.binary_image_resolved, 'pull_specification', None),
+            'bundles': [bundle.pull_specification for bundle in self.bundles],
+            'from_index': getattr(self.from_index, 'pull_specification', None),
+            'from_index_resolved': getattr(self.from_index_resolved, 'pull_specification', None),
+            'index_image': getattr(self.index_image, 'pull_specification', None),
+            'state': _state_to_json(self.state),
```

Comment:
I don't know what SQLAlchemy does under the hood, but it might be nice to do this as part of the else branch of the `if verbose`. The `if verbose` branch can then pick the 0th index of `states` to get the latest state. It might save one query to the database if `verbose` is `True`.